### DISCUSSION TITLE
bottle lid robot manipulation

### DISCUSSION THREAD

**Discussion by shivanimall** [haosulab/SAPIEN#76] [Q&A] [answered]
Hello, I saw that you have a bottle with lid revolute and prismatic modes. https://sapien.ucsd.edu/browse

Do you have an example demonstrating lid revolute robot manipulation behaviour? i.e. robot arm revolving a lid

Thanks for your help.

**Comment by fbxiang** [comment, marked as answer]
The lid movement is not currently supported in SAPIEN, and I do not know an existing simulator that is able to simulate this movement easily and reliably. So currently, this annotation is really limited to visual rendering only.

**Reply by shivanimall** [reply]
Many thanks for the response @fbxiang

**Comment by shivanimall** [comment]
Also @fbxiang are there any examples [here](https://sapien.ucsd.edu/browse) that are for rendering purposes and may not be able to simulate for manipulation behaviour? For example door knob revolute, vault / safe knob revolute

**Reply by fbxiang** [reply]
Only screw joints (coupled revolute and prismatic) cannot be simulated (e.g. bottle lid). They do not even have a representation in the URDF format! So when we export URDF, we split the revolute and prismatic into 2 separate joints.
All revolute and prismatic joints can be simulated in most simulators. However, most simulators will only handle convex collision shapes for dynamic objects (some simulators do offer non-convex collision but they are typically very slow), but our shapes are not convex. So depending on use cases, you may want to pre-process the collision shapes with some algorithms such as VHACD.

**Reply by shivanimall** [reply]
Many thanks @fbxiang for sharing, that is very helpful.
